### PR TITLE
Fix managers and groups link

### LINKED ISSUE #2932: Manager unable to edit collections on testing branch
### Discussed in https://github.com/dani-garcia/vaultwarden/discussions/2930

<div type='discussions-op-text'>

<sup>Originally posted by **louisfgr** November 23, 2022</sup>
When trying to change or add new collections in the testing branch (v 1.26.0-7a767310) as a manager, the user is logged out and the following error appears in the log.

The issue seems to be similar to #2151 from Dec 2021

```
[2022-11-23 12:42:56.720][auth][ERROR] Unauthorized Error: You need to be Admin or Owner to call this endpoint
[2022-11-23 12:42:56.721][_][WARN] Request guard `AdminHeaders` failed: "You need to be Admin or Owner to call this endpoint".
[2022-11-23 12:42:56.722][_][WARN] No 401 catcher registered. Using Rocket default.
```

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.26.0-7a767310
* Web-vault version: v2022.10.2
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.39.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": false,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*************.********.***",
  "domain_origin": "*****://*************.********.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": false,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "BitwardenTest",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "warn",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": false,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "********.***",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*************@********.***",
  "smtp_from_name": "BitwardenTest",
  "smtp_host": "***.***.**.**",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_security": "off",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": 14,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details></div>

### PR DESCRIPTION
**Pull request by BlackDex**
This PR should fix the managers and group link.
Although i think there might be a cleaner sollution, there are a lot of other items to fix here which we should do in time.

But for now, with theh group support already merged, this fix should at least help solving issue #2932.

Fixes #2932

I know, it can be nicer, but lets just leave it like this, and fix all the items later :) .

### CODE REVIEW

**Review by stefan0xC — COMMENTED** [review]
I can confirm that it works but I am not sure about all the joins, the naming of the functions and where the functions should be located.

Maybe it would be better to have multiple has_access_ functions (i.e. in addition to `has_full_access`, we could have a `has_direct_access_to_collection`, `is_in_group_with_full_access` and `is_in_group_with_access_to_collection`)? That would make it hopefully easier to reason with.

Given that we already get the `OrgHeaders` from the path I think some lookups could also probably be improved, i.e. `is_in_group_with_access_to_collection` could check if there is an intersection between collections_users (given the userorg id) and groups_collections (given the collection id) on the group id.

File: src/db/models/collection.rs [hunk 1]
```diff
@@ -203,6 +203,13 @@ impl Collection {
         }}
     }
 
+    // Check if a user has access to a specific collection
+    // FIXME: This needs to be reviewed. The query used by `find_by_user_uuid` can also be improved
+    //        It is probably best to create separate query for this, but for now this is a good solution.
+    pub async fn find_by_collection_and_user_uuid(collection_uuid: &str, user_uuid: &str, conn: &mut DbConn) -> bool {
```
[the remark this answers is not in the document]
Comment (BlackDex) [reply src/db/models/collection.rs]: You are right.

**Comment by BlackDex** [comment]
> If this is the only PR that blocks a new release, improvements should be made after the new release.
> 
> Update: Bitwarden has updated all their clients. I am just a bit nervous that people won't be able to log into 1.26.0 anymore, if they auto-update their clients or even update by accident.

FYI: This is an Open Source project, maintained by people in there sparse free time, not a Bitwarden Inc. who has people on there payroll.

**Comment by tessus** [comment]
> For me it felt/read like, hurry up with a new release already.

Nope, not at all. At least not for me (as I have mentioned seeral times). It was a mere reminder that you might get support issues in gh/forum/matrix soon. From a logistic standpoint it makes a lot more sense to release a version that works with the current clients. Fixing a feature that was introduced and does not have to be used is secondary. Or is this a wrong assessment?

I have created rpm packages, which I can and will install as soon as one of my users notifies me that they can't login or face other issues with the new clients.

> Then why the remark about it at all?

I also only mentioned it, because you wrote that sentence in the first post. You yourself seemed to be eager to get this merged and get out a new release. Otherwise why would you have written that sentence?

Anyway, don't worry, I won't mention it ever again.

**Comment by BlackDex** [comment]
> > For me it felt/read like, hurry up with a new release already.
> 
> Nope, not at all. At least not for me (as I have mentioned seeral times). It was a mere reminder that you might get support issues in gh/forum/matrix soon. From a logistic standpoint it makes a lot more sense to release a version that works with the current clients. Fixing a feature that was introduced and does not have to be used is secondary. Or is this a wrong assessment?
> 

Well this pr fixes a new feature which people will use, and later in will cause issues also.

> I have created rpm packages, which I can and will install as soon as one of my users notifies me that they can't login or face other issues with the new clients.
> 
> > Then why the remark about it at all?
> 
> I also only mentioned it, because you wrote that sentence in the first post. You yourself seemed to be eager to get this merged and get out a new release. Otherwise why would you have written that sentence?
> 

That was more because i think we could do some more assessments on all the queries and functions linked to it. Maybe combine, or split etc.. it was more a note/reminder for the future to work on that.

> Anyway, don't worry, I won't mention it ever again.

Anything productive or stuff that questions something is always allowed, that will hopefully better Vaultwarden and any other project.

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**